lib/core/src/lib/datatable/components/datatable/datatable.component.spec.ts

```diff
@@ -563,25 +563,33 @@ describe('DataTable', () => {
         expect(rows[1].isSelected).toBeTruthy();
     });
 
-    it('should NOT unselect the row with [single] selection mode', (done) => {
+    it('should unselect the row with [single] selection mode', (done) => {
         dataTable.selectionMode = 'single';
         dataTable.data = new ObjectDataTableAdapter(
             [
-                { name: '1', isSelected: true },
+                {name: '1'},
                 { name: '2' }
             ],
             [new ObjectDataColumn({ key: 'name' })]
         );
         const rows = dataTable.data.getRows();
         dataTable.ngOnChanges({});
         fixture.detectChanges();
-
-        dataTable.rowClick.subscribe(() => {
+        dataTable.onRowClick(rows[0], new MouseEvent('click'));
+        dataTable.rowClick.pipe(
+            take(1)
+        ).subscribe(() => {
             expect(rows[0].isSelected).toBeTruthy();
             expect(rows[1].isSelected).toBeFalsy();
-            done();
+            dataTable.onRowClick(rows[0], new MouseEvent('click'));
+            dataTable.rowClick.pipe(
+                take(1)
+            ).subscribe(() => {
+                expect(rows[0].isSelected).toBeFalsy();
```

Comment:
If checking for booleans please use `toBeFalse` or `toBeTrue` as `toBeFalsy` would also accept other falsy values like `undefined`